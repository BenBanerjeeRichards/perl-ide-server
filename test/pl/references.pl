$x->{hello}{world};
{ hello };
$y->{ world }{"test"};
$x->{"hello"}{"world"};
$x->{$code};
$x->{  $code };
$x->{ $main::test::'a};
