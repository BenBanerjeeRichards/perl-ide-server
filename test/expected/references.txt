ScalarVariable($x)
Operator(->)
HashDerefStart({)
HashKey(hello)
HashDerefEnd(})
HashDerefStart({)
HashKey(world)
HashDerefEnd(})
;
LBracket
Name(hello)
RBracket
;
ScalarVariable($y)
Operator(->)
HashDerefStart({)
HashKey(world)
HashDerefEnd(})
HashDerefStart({)
StringStart(")
String(test)
StringEnd(")
HashDerefEnd
;
ScalarVariable($x)
Operator(->)
HashDerefStart({)
StringStart(")
String(hello)
StringEnd(")
HashDerefEnd
HashDerefStart({)
StringStart(")
String(world)
StringEnd(")
HashDerefEnd
;
ScalarVariable($x)
Operator(->)
HashDerefStart({)
ScalarVariable($code)
HashDerefEnd
;
ScalarVariable($x)
Operator(->)
HashDerefStart({)
ScalarVariable($code)
HashDerefEnd
;
ScalarVariable($x)
Operator(->)
HashDerefStart({)
ScalarVariable($main::test::'a)
HashDerefEnd
;
EndOfInput
